modules/tlsinfo.cpp

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (C) 2015 Michael Morris
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Notes: */
+/* This module sends the certificate chain of a connection to the
+ connected client so that the client can display this information
+ in a user friendly dialog. */
+/* As-is customary, the chain starts with the peer certificate and
+ and ends with the root certificate authority certificate. */
+/* The certificate chain is sent using the BATCH command. */
+/* Each certificate is surrounded in its own nested BATCH within
+ a single global BATCH for the whole session. */
+
+/* Example response:
+
+	>> :znc.in BATCH +1f5fdd znc.in/tlsinfo
+	>> @batch=1f5fdd :znc.in BATCH +6f7a1a znc.in/tlsinfo-certificate
+	>> @batch=6f7a1a :znc.in CERTINFO ExampleUser :-----BEGIN CERTIFICATE-----
+	>> @batch=6f7a1a :znc.in CERTINFO ExampleUser :...
+	>> @batch=6f7a1a :znc.in CERTINFO ExampleUser :-----END CERTIFICATE-----
+	>> @batch=1f5fdd :znc.in BATCH -6f7a1a
+	>> @batch=1f5fdd :znc.in BATCH +31fbfc znc.in/tlsinfo-certificate
+	>> @batch=31fbfc :znc.in CERTINFO ExampleUser :-----BEGIN CERTIFICATE-----
+	>> @batch=31fbfc :znc.in CERTINFO ExampleUser :...
+	>> @batch=31fbfc :znc.in CERTINFO ExampleUser :-----END CERTIFICATE-----
+	>> @batch=1f5fdd :znc.in BATCH -31fbfc
+	>> :znc.in BATCH -1f5fdd
+ */
+
+#include <znc/Modules.h>
+#include <znc/IRCNetwork.h>
+#include <znc/IRCSock.h>
+
+static const char *TlsInfoCap = "znc.in/tlsinfo";
+
+static const char *TlsInfoBatchGlobalType = "znc.in/tlsinfo";
+static const char *TlsInfoBatchChildType = "znc.in/tlsinfo-certificate";
+
+class CTlsInfoMod : public CModule
+{
+public:
+
+#ifndef HAVE_LIBSSL
+	MODCONSTRUCTOR(CTlsInfoMod) {}
+
+	bool OnLoad(const CString&, CString& sMessage) {
+		sMessage = "Module built against install of ZNC that lacks SSL support.";
```

Comment:
Other modules do it by `#define REQUIRESSL 1`
Without SSL, such modules are not built by the build system.